django/core/serializers/base.py

```diff
@@ -336,7 +336,7 @@ def build_instance(Model, data, db):
         and hasattr(default_manager, "get_by_natural_key")
         and hasattr(Model, "natural_key")
     ):
-        natural_key = Model(**data).natural_key()
+        natural_key = Model.from_db(db, data.keys(), data.values()).natural_key()
```

Comment:
`_state` is not really a private API, it's documented in multiple places. Also `from_db()` [can be overwritten](https://docs.djangoproject.com/en/4.0/ref/models/instances/#customizing-model-loading) and set `adding` to `False` instead of `True`, so I'd prefer a solution with the fewer possible side-effects.